crates/cheatcodes/src/utils.rs

```diff
@@ -372,4 +393,81 @@ mod tests {
         let result = sign_p256(&U256::ZERO, &digest, &mut cheats);
         assert_eq!(result.err().unwrap().to_string(), "private key cannot be 0");
     }
+
+    #[test]
+    fn test_affine_coordinates_p256() {
+        let pk_u256: U256 = "100".parse().unwrap(); // Parse private key, assuming valid U256 creation
+        let result = generate_public_key_p256(&pk_u256).unwrap(); // Get the ABI-encoded result
+
+        // Ensure the result has exactly 64 bytes (32 for x and 32 for y)
+        assert_eq!(result.len(), 64, "Resulting byte array must be exactly 64 bytes long.");
+
+        // Convert Vec<u8> to a fixed-size array of 64 bytes
+        let result_bytes: [u8; 64] = result.try_into().expect("Slice with incorrect length");
+
+        // Extract x and y coordinates from the array
+        let x_bytes: [u8; 32] = result_bytes[0..32].try_into().expect("Error slicing x bytes");
+        let y_bytes: [u8; 32] = result_bytes[32..64].try_into().expect("Error slicing y bytes");
+
+        // Convert byte arrays into U256 values
+        let x_coord = U256::from_be_bytes(x_bytes);
+        let y_coord = U256::from_be_bytes(y_bytes);
+
+        // Use x_coord and y_coord as U256 values in your tests
+        println!("x_coord: {x_coord}");
+        println!("y_coord: {y_coord}");
+    }
+
+    #[test]
+    fn test_p256_publickey_generation() {
+        // Hexadecimal private key
+        let hex_private_key = "8449b2b5f4dbeaf8b454fe16cb1f577c3d05df5224021390438b5c2c5a8ac235";
+
+        // Convert the hex string to bytes
+        let private_key_bytes = hex::decode(hex_private_key).expect("Decoding failed");
+
+        // Ensure the byte array is of correct length (32 bytes for P-256)
+        assert_eq!(private_key_bytes.len(), 32, "Private key must be 32 bytes long");
+
+        // Convert the byte array to a fixed-size array
+        let private_key_array: [u8; 32] =
+            private_key_bytes.try_into().expect("Invalid length for private key");
```

Comment:
all of this is `b256!("8449b2b5f4dbeaf8b454fe16cb1f577c3d05df5224021390438b5c2c5a8ac235")`